### PR TITLE
Quickstart Solution using devops tool - Jenkins, Chef Habitat & Terraform

### PR DESCRIPTION
**Pull request by nvtuluva**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,391 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: remove all of the defaultValues that are not valid...

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+        },
+        "buildInstanceVmSize": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "vm size"
+            }
+        },
+        "azureApplicationId": {
+            "type": "securestring",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: this does not need to be secure

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+            }
+        },
+        "azureClientSecret": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "ad clientsecret"
+            }
+        },
+        "_artifactsLocation": {
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: see the BP guide for how to handle artifacts - you need a defaultValue and a sasToken

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+        },
+        "azureSubscriptionId":"[subscription().subscriptionId]",
+        "azureTenantId":"[subscription().tenantId]",
+        "resourceGroupName": "[resourceGroup().name]",
+        "packerStorageAccountName": "[concat('packerstrg',variables('suffix'))]",
+        "publicIpAddressType": "Dynamic",
+        "location": "[resourceGroup().location]",
+        "deploymentApiVersion": "2016-02-01",
+        "suffix": "[substring(uniqueString(resourceGroup().id), 0, 5)]",
+        "baseUrl": "[parameters('_artifactsLocation')]"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: remove this and just use the parameter

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+        },
+        "JenkinsFQDN": {
+            "type": "string",
+            "value": "[reference('Jenkins').outputs.jenkinsDNS.value]"
+        },
+        "JenkinsWebUIURL": {
+            "type": "string",
+            "value": "[concat(reference('Jenkins').outputs.jenkinsDNS.value,':8080')]"
+        },
+        "kibanaWebUIUsername": {
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: why are we hardcoding outputs... particularly secrets?

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "value": "adminuser"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: GEN-UNIQUE

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        "authenticationType": {
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: remove this and use a defaultValue

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+            "value": "sshPublicKey"
+        },
+        "password":{
+            "value": "Password@1234"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: GEN-PASSWORD

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        "sshPublicKey": {
+            "value": "ssh-rsa AAAAB3NzaC1yc2EAAAABJQAAAQEAr8HZ2JCXEb5Dpd3VmaxK/PE4/nu6mULEr7KfYBQdtDsjfLy7l+flHBwaRqo6LzFh+VTf55iHjAioJCJPLo5XJxIxIqZxpxdiNJWS9ai1dVtM+nOrzexyUNNovuATqBHLikvCGy4E0TeObNjIOIei70c+LaGUiHOf6KUe6KB3zbRRxtbL/s4XPWsVBuxwCsEGQkWl41HAruZRTf6pGdtopxL27vWd8nafJxf0jPFh9WVTC0UlgfOGrOt6UKIaJc+Q1bE7apDdd0Nqrx8ewDq8soJ6rOkpTwc4VrDifHsT0/vqwF88MYkItbwYMOYrlURINrUM1SHVsI9uy6toAyBWtw== rsa-key-20170912"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: GEN-SSH-PUBLIC-KEY

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        "jenkinsVmSize": {
+            "value": "Standard_DS2_v2"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: use defaultValue

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        "buildInstanceVmSize": {
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: use defaultValue

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+            "value": "Standard_DS1_v2"
+        },
+        "azureApplicationId": {
+            "value": "GEN-UNIQUE"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: GEN-GUID

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+        },
+        "azureClientSecret": {
+            "value": "GEN-UNIQUE"
+        },
+        "_artifactsLocation":{
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: see BP guide for handling artifacgts

File: devopstools-jenkins-chefhabitat-terraform/nested/README.md [hunk 14]
```diff
@@ -0,0 +1 @@
+
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/README.md]: remove this file

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+                "networkSecurityGroup": {
+                    "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('buildInstanceSettings').buildInstanceNsgName)]"
+                }
+            }
+        },
+        {
+            "comments": "This is the virtual machine resource of build instance.",
+            "name": "[parameters('buildInstanceSettings').buildInstanceVmName]",
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: update apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+                        "storageUri": "[reference(resourceId('Microsoft.Storage/storageAccounts', parameters('buildInstanceSettings').buildInstanceDiagnosticsStorageAccountName)).primaryEndpoints.blob]"
+                    }
+                }
+            }
+        },
+        {
+            "comments": "This custom script extension is used to install jenkins.",
+            "name": "[concat(parameters('buildInstanceSettings').buildInstanceVmName, '/buildscript')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: update apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+        "AzureTenantId": {
+            "type": "string"
+        },
+        "packerStorageAccName":{
+            "type": "string"
+        }
+    },
+    "variables": {
+        "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('networkSettings').virtualNetworkName)]",
+        "subnetRef": "[concat(variables('vnetId'), '/subnets/', parameters('networkSettings').subnet1Name)]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: simplify - just use resourceId

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+            "type": "string"
+        },
+        "AzureSubscriptionId": {
+            "type": "string"
+        },
+        "AzureApplicationId": {
+            "type": "string"
+        },
+        "AzureClientSecret": {
+            "type": "string"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json]: securestring

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+            "type": "object"
+        },
+        "AzureSubscriptionId": {
+            "type": "string"
+        },
+        "AzureApplicationId": {
+            "type": "string"
+        },
+        "AzureClientSecret": {
+            "type": "string"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: securestring

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json [hunk 20]
```diff
@@ -0,0 +1,56 @@
...
+        "packerStorageAccountName": {
+            "type": "string"
+        }
+    },
+    "resources": [
+        {
+            "comments": "This custom script extension is used to create jobs in jenkins.",
+            "name": "[concat(parameters('jenkinsSettings').jenkinsVmName, '/Init')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json]: update apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json [hunk 21]
```diff
@@ -0,0 +1,239 @@
...
+            "type": "string"
+        },
+        "AzureSubscriptionId": {
+            "type": "string"
+        },
+        "AzureApplicationId": {
+            "type": "string"
+        },
+        "AzureClientSecret": {
+            "type": "string"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: securestring

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json [hunk 22]
```diff
@@ -0,0 +1,239 @@
...
+        },
+        "AzureTenantId": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "sasToken": "",
+        "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('networkSettings').virtualNetworkName)]",
+        "subnetRef": "[concat(variables('vnetId'), '/subnets/', parameters('networkSettings').subnet1Name)]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: just use resourceId

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json [hunk 23]
```diff
@@ -0,0 +1,239 @@
...
+                        "storageUri": "[reference(resourceId('Microsoft.Storage/storageAccounts', parameters('jenkinsSettings').jenkinsDiagnosticsStorageAccountName)).primaryEndpoints.blob]"
+                    }
+                }
+            }
+        },
+        {
+            "comments": "This custom script extension is used to install jenkins.",
+            "name": "[concat(parameters('jenkinsSettings').jenkinsVmName, '/Init')]",
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "apiVersion": "2016-04-30-preview",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: update apiVersion

File: devopstools-jenkins-chefhabitat-terraform/scripts/README.md [hunk 24]
```diff
@@ -0,0 +1 @@
+
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/scripts/README.md]: remove this file

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json (hunk 1 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: Removed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json (hunk 2 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: Updated to string

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json (hunk 4 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json (hunk 5 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 6 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-UNIQUE

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 7 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Removed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 8 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-PASSWORD

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 9 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: changed to GEN-SSH-PUBLIC-KEY

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 10 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Updated

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 11 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Updated

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 12 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-GUID

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 6 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-UNIQUE

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 7 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Removed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 8 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-PASSWORD

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 9 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-SSH-PUBLIC-KEY

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 10 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Updated

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 11 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Updated

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 12 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: Changed to GEN-GUID

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json (hunk 13 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: updated

File: devopstools-jenkins-chefhabitat-terraform/nested/README.md (hunk 14 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/README.md]: Removed

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json (hunk 19 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json (hunk 17 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json (hunk 15 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: Used latest apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json (hunk 16 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: Used latest apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json (hunk 18 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json (hunk 20 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/jenkins-jobs.json]: Used latest apiVersion

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json (hunk 21 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json (hunk 22 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: Fixed

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json (hunk 23 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: Used latest apiVersion

File: devopstools-jenkins-chefhabitat-terraform/scripts/README.md (hunk 24 above)
Comment (SrikalaRekapalli) [reply devopstools-jenkins-chefhabitat-terraform/scripts/README.md]: Removed

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+        "kibanaWebUIPassword": {
+            "value": "GEN-PASSWORD"
+        },
+        "azureApplicationId": {
+            "value": "GEN-GUID"
+        },
+        "azureClientSecret": {
+            "value": "GEN-UNIQUE"
+        },
+        "_artifactsLocation": {
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.parameters.json]: remove this and rely on the defaultValue

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 26]
```diff
@@ -0,0 +1,385 @@
...
+            "buildInstanceDnsLabelPrefix": "[concat('buildserver',variables('suffix'))]",
+            "buildInstanceNsgName": "build-nsg",
+            "buildInstanceNicName": "build-nic",
+            "buildInstanceVmName": "buildinstance",
+            "adminUsername": "[parameters('adminUsername')]",
+            "authenticationType": "[parameters('authenticationType')]",
+            "sshPublicKey": "[parameters('sshPublicKey')]",
+            "buildInstanceVmSize": "[parameters('buildInstanceVmSize')]",
+            "ubuntuSku": "16.04-LTS",
+            "buildScriptUrl": "[concat(parameters('_artifactsLocation'),'/scripts/build.sh')]"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: this needs the sasToken appended

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 27]
```diff
@@ -0,0 +1,385 @@
...
+            "kibanaWebUIUsername":"[parameters('kibanaWebUIUsername')]",
+            "kibanaWebUIPassword":"[parameters('kibanaWebUIUsername')]",
+            "authenticationType": "[parameters('authenticationType')]",
+            "sshPublicKey": "[parameters('sshPublicKey')]",
+            "jenkinsVmSize": "[parameters('jenkinsVmSize')]",
+            "ubuntuSku": "16.04-LTS",
+            "storageAccountType": "Standard_LRS",
+            "installJenkinsScriptName": "install_jenkins.sh",
+            "jenkinsReleaseType": "LTS",
+            "baseUrl": "[parameters('_artifactsLocation')]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: this is not used

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 28]
```diff
@@ -0,0 +1,385 @@
...
+            "installJenkinsScriptUrl": "[concat(parameters('_artifactsLocation'),'/scripts/install_jenkins.sh')]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: need to append sasToken

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 29]
```diff
@@ -0,0 +1,385 @@
...
+            "installJenkinsjobsScriptUrl": "[concat(parameters('_artifactsLocation'),'/scripts/jenkins_deploy.sh')]"
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: append sasToken

File: devopstools-jenkins-chefhabitat-terraform/azuredeploy.json [hunk 30]
```diff
@@ -0,0 +1,385 @@
...
+            "ubuntuSku": "16.04-LTS",
+            "buildScriptUrl": "[concat(parameters('_artifactsLocation'),'/scripts/build.sh')]"
+        },
+        "azureSubscriptionId":"[subscription().subscriptionId]",
+        "azureTenantId":"[subscription().tenantId]",
+        "resourceGroupName": "[resourceGroup().name]",
+        "packerStorageAccountName": "[concat('packerstrg',variables('suffix'))]",
+        "publicIpAddressType": "Dynamic",
+        "location": "[resourceGroup().location]",
+        "deploymentApiVersion": "2016-02-01",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/azuredeploy.json]: remove unused variables

File: devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json [hunk 31]
```diff
@@ -0,0 +1,216 @@
...
+        },
+        "AzureTenantId": {
+            "type": "string"
+        },
+        "packerStorageAccName":{
+            "type": "string"
+        }
+    },
+    "variables": {
+        "subnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', parameters('networkSettings').virtualNetworkName), '/subnets/', parameters('networkSettings').subnet1Name)]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/build-instance.json]: remove concat and just use resourceId

File: devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json [hunk 32]
```diff
@@ -0,0 +1,226 @@
...
+        "AzureClientSecret": {
+            "type": "securestring"
+        },
+        "AzureTenantId": {
+            "type": "string"
+        }
+    },
+    "variables": {
+        "sasToken": "",
+        "subnetRef": "[concat(resourceId('Microsoft.Network/virtualNetworks', parameters('networkSettings').virtualNetworkName), '/subnets/', parameters('networkSettings').subnet1Name)]",
```
Comment (bmoore-msft) [inline devopstools-jenkins-chefhabitat-terraform/nested/jenkins.json]: remove concat and just use resourceId

**Pull request merged**